addon/services/concurrent-axe.js

```diff
@@ -0,0 +1,38 @@
+/* global axe */
+import Service from '@ember/service';
+import { next } from '@ember/runloop';
+
+export default Service.extend({
+  init() {
+    this._super(...arguments);
+    this._timer = null;
+    this._queue = [];
+  },
+
+  /**
+   * Axe v3 contains a concurrency issue which breaks the component auditing feature.
+   * This service defers axe.run calls on onto the next loop so that concurrent 
+   * axe executions do not occur.
+   *
+   * @see(https://github.com/dequelabs/axe-core/issues/1041)
+   * @public
+   * @param {HTMLElement} element axe context
+   * @param {Object} options axe configuration options
+   * @param {Function} callback axe audit callback
+   * @return {Void}
+   */
+  run(element, options, callback) {
+    if (this._timer) {
+      this._queue.push(arguments);
+    } else {
+      this._timer = next(() => {
```

Comment:
I'm slightly concerned about the async added here. For example:

* What happens when the element is removed from the DOM before the audit has occurred? Does axe care if the element that is audited is in-dom? 
* How do we ensure the element state is consistent (between when the audit was started and when it is ran)?

Reply:
Removal of the component root element can definitely be problematic. As such, I've added a check to ensure that the element is still in the DOM before it is passed off to aXe (and also updated corresponding tests). Unfortunately, ensuring consistent element state is problematic even without the async, as the state can change between when aXe initially starts the audit on a component and when it completes. This concern is somewhat negligible given the nature of the tool - as it's well understood it will not catch 100% of all accessibility violations. A few milliseconds between operations would likely not incur drastic state changes, especially given that this is for developer use (I'd hope other developers wouldn't be interacting with a component before its corresponding audit completed). Additionally, the audit would rerun on each subsequent re-render.

Reply:
Ok, seems good enough to me 😃. We can always circle back and attempt to address if there is any actual issue here.

Reply:
Yes, thanks Robert!